src/dba/AbstractModelFactory.class.php

```diff
@@ -872,7 +873,8 @@ public function getDB($test = false) {
       if ($test) {
         return null;
       }
-      die("Fatal Error ! Database connection failed: " . $e->getMessage());
+      UI::printError(UI::ERROR, "Fatal Error! Database connection failed: " . $e->getMessage());
```

Comment:
It might be worth to consider if $e->getMessage() can be omitted from the message but written to a log instead.
By doing so the error message stays more generic but can still be debugged (good practice) albeit optional.

Reply:
So far there is no file-based log on the server side which would be suitable for such messages.
I don't really see a big benefit from moving the error message to some extra location (which users again first need to be aware of).

Reply:
I would agree that a log file would be better, but we do not really have a server log file; so this is a nice solution as for now.